vm/plugin-context.cpp

```diff
@@ -497,7 +497,6 @@ PluginContext::popTrackerAndSetHeap()
 int
 PluginContext::pushTracker(uint32_t amount)
 {
-  assert(usesHeapTracker());

```

Comment:
The tracker must not be used in >= 1.11 binaries. Can you clarify what is being fixed here? I didn't see an explanation or test case.

Reply:
The problem is HeapPop doesn't deallocate the resource allocated by HeapAlloc2dArray

https://github.com/alliedmodders/sourcemod/blob/fix-menu-vote-handler/core/logic/smn_menus.cpp#L510

Returns error 5. SP_ERROR_INVALID_ADDRESS.

Reply:
![image](https://user-images.githubusercontent.com/20559490/142671347-8399d932-1144-4bad-84c2-b21501c10dd9.png)
This is the error from mapchooser_extended after running a while

Reply:
I didn't know the tracker usage.

Reply:
Okay, thanks. This API does feel very weird to me. I'll add an RAII thing (eg "AutoHeapScope") to make this way easier to use. The current API is too difficult.